File: Data Structures & Algorithms/remove-element/submission-4.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        n = len(nums)
        i = 0
        while i < n:
            if nums[i] == val:
                n -= 1
                nums[i], nums[n] = nums[n], nums[i]
                i -= 1
            i += 1
        return n

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-4.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        std::unordered_map<int, int> map;
        map[0] = 1;
        int sum = 0;
        int cnt = 0;
        for (int x : nums) {
            sum += x;
            int remaining = sum - k;
            cnt += map[remaining];
            map[sum]++;
        }
        return cnt;
    }
};